src/components/header/header.js

```diff
@@ -1,45 +1,33 @@
-import React, { useState, useEffect, useContext, useMemo, useCallback } from 'react';
+import React, { useState, useEffect, useMemo, useCallback } from 'react';
 import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
 import Logo from '../logo';
 import Search from '../search';
 import ModalWindow from '../modalWindows/modalWindow';
 import AddMovie from '../modalWindows/addMovie';
 import MovieDetails from '../movieDetails';
-import MockDataContext from '../../context';
+import { movieShowDetails, movieHideDetails } from './../../actions/actions';
 import { Button } from 'react-bootstrap';
 
 import './header.css'
 
-const useShowMovieDetails = (id, func) => {
-    useEffect(
-        () => {
-            if(id !== ''){
-                func(true);
-            }
-    }, [id]);
-}
-
 const Header = (props) => {
     const [show, setShow] = useState(false);
-    const [showDetails, setShowDetails] = useState(false);
     
     const handleClose = useCallback(() => setShow(false), []);
     const handleShow = useCallback(() => setShow(true), []);
-    const handleCloseDetails = useCallback(() => setShowDetails(false), []);
-    
-    const movies = useContext(MockDataContext);
+    const handleCloseDetails = useCallback(() => hideDetails(), []);
```

Comment:
`showDetails` is used only this component, it's not best way to save it to `redux` store

Reply:
done